src/client/Microsoft.Identity.Client/ApiConfig/Parameters/AcquireTokenByAuthorizationCodeParameters.cs

```diff
@@ -11,7 +11,7 @@ internal class AcquireTokenByAuthorizationCodeParameters : IAcquireTokenParamete
 
         public string PkceCodeVerifier { get; set; }
 
-        public bool SendX5C { get; set; }
+        public bool? SendX5C { get; set; }
```

Comment:
>? [](http://example.com/codeflow?start=19&length=1)

i don't think this will work in MSAL, they don't support the nullable reference types introduced in c# 8.

Reply:
Help me understand why nullable bool is required and not just bool?

Reply:
@jennyf19 - nullable types were introduced in very early C# version, we do support them in MSAL. However, nullable *reference* types are not supported because of C# 8 - https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references

Reply:
That's what I thought I wrote. `nullable reference types`

Reply:
Ah, darn.  This makes things much trickier - I'm worried that without being nullable it will default to false on a per-request basis, and overwrite the cca field that should be the primary driver of the value. 

Goal here: If the request params explicitly has SendX5C set, use that, otherwise fall back to the default-false cca object. Currently I'm using the nullable as a way to signal it's been explicitly set.  Should I add a new field instead, that's set during the per-request call, something like `bool overrideCcaSendX5C` in `IAcquireTokenParameters`?

Reply:
Sorry, I mean that nullable **value** types was introduced since c# 2.0. Nullable reference types was introduced in C# 8.0

![image](https://user-images.githubusercontent.com/12273384/127895369-8a55ef1e-c258-4c6d-a8b1-c7d6659211eb.png)

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-value-types

Reply:
@hpsin I think it would be better to modify the logic internally to choose the proper value based on the parameters set rather than adding additional api here.

Reply:
Can you clarify exactly what you mean by modify the logic internally? I don't thing I have the same name references for things that your team does :(  The specific line you're commenting on makes the notion of setting X5C per-request nullable so that we can perform internal logic to figure out what to do, rather than adding fields to the request parameters object. We can't do this without an API in general, as the developer needs to have some way of specifying whether we do this.